src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -249,124 +249,128 @@ public RTPEncodingDesc(
+     * newerFrame, assuming olderFrame predates/is older than newerFrame.
+     * Depending on the relationship of olderFrame and newerFrame, and what
+     * we know about olderFrame and newerFrame, we may be able to deduce
+     * the last expected sequence number for olderFrame and/or the first
+     * expected sequence number of newerFrame.
      *
-     * @param a the {@link FrameDesc} that comes before b.
-     * @param b the {@link FrameDesc} that comes after a.
+     * @param olderFrame the {@link FrameDesc} that comes before newerFrame.
+     * @param newerFrame the {@link FrameDesc} that comes after olderFrame.
      */
-    private static void applyFrameBoundsHeuristics(FrameDesc a, FrameDesc b)
+    private static void applyFrameBoundsHeuristics(
+            FrameDesc olderFrame,
+            FrameDesc newerFrame)
     {
-        int aLastSeqNum = a.getEnd(), bFirstSeqNum = b.getStart();
-        if (aLastSeqNum != -1 && bFirstSeqNum != -1)
+        if (olderFrame.lastSequenceNumberKnown() && newerFrame.lastSequenceNumberKnown())
         {
-            // No need for heuristics.
+            // We already know the last sequence number of olderFrame and the first
+            // sequence number of newerFrame, no need for further heuristics.
             return;
         }
-
-        long tsDiff = (b.getTimestamp() - a.getTimestamp()) & 0xFFFFFFFFL;
-        if (tsDiff > (1L << 30) && tsDiff < (-(1L << 30) & 0xFFFFFFFFL))
+        if (!TimestampUtils.isNewerTimestamp(
+                newerFrame.getTimestamp(), olderFrame.getTimestamp()))
         {
-            // the distance (mod 2^32) between the two timestamps needs to be
-            // less than half the timestamp space.
+            // newerFrame isn't newer than olderFrame, bail
             return;
         }
-        else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
-        {
-            logger.warn("Frames that are out of order detected.");
-        }
-        else
+        int lowestSeenSeqNumOfNewerFrame = newerFrame.getMinSeen();
+        int highestSeenSeqNumOfOlderFrame = olderFrame.getMaxSeen();
+        int seqNumDiff =
+                RTPUtils.sequenceNumberDiff(lowestSeenSeqNumOfNewerFrame, highestSeenSeqNumOfOlderFrame);
+
+        boolean guessed = false;
+
+        // We have some codecs (VPX) for which we can inspect the payload
+        // to deduce the start/end of olderFrame frame. This has an impact on the
+        // distances between the frames that allow us to guess frame
+        // boundaries.
+        //FIXME(brian): why does this allow us to infer anything extra from the sequence number
+        // differences?  we don't leverage frame boundary markings in any way when we deduce
+        // first/last sequence numbers.  does it just mean that these frames would've already
+        // (conclusively) had the start/end seq numbers set?  if so, we wouldn't need to
+        // call this method at all for those frames, right?
```

Comment:
I'm sorry, I'm not following the discussion. But this code is not designed to leverage Picture IDs, it's designed to leverage frame markings which are available in the VPX payload descriptor and in the frame markings header extension. @bgrozev suggested the use of VPX picture IDs and I like the idea. 

>> why does this (frame markings support) allow us to infer anything extra from the sequence number differences?

Here's an example that hopefully clarifies things: First suppose that we get two frames that their diff is 3 and we haven't seen the start nor the end (n.b. we can always see the end by inspecting the RTP marker bit), e.g. we get this sequence [0, 3] with 0 from older frame and 3 from the newer frame.

Now, if both newer and older frames support frame markings, then we know there must be a start and an end in between 0 and 3, and since the only available slots are 1 and 2, 1 must be the end and 2 must be the start.

Unfortunately, if frame markings are not supported, then we can't deduce neither the start nor the end, because the end could be 1 or 2 (not 0 because it doesn't have the RTP marker bit set), and the start could be 2 or 3.